ArchiSteamFarm/Bot.cs

```diff
@@ -2194,6 +2195,32 @@ public sealed class Bot : IDisposable {
 			await Actions.AcceptGuestPasses(guestPassIDs).ConfigureAwait(false);
 		}
 
+		private bool ShouldMarkGroupChatMessage(ulong steamID) {
+			if (BotConfig.BotBehaviour.HasFlag(BotConfig.EBotBehaviour.MarkReceivedMessagesAsRead)) {
+				return true;
+			}
+
+			if (BotConfig.BotBehaviour.HasFlag(BotConfig.EBotBehaviour.MarkBotMessagesAsRead) &&
+				Bots.Values.Any(bot => bot.SteamID == steamID)) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private bool ShouldMarkPrivateChatMessage(ulong steamID, string message) {
+			if (ShouldMarkGroupChatMessage(steamID)) {
+				return true;
+			}
+
+			if (BotConfig.BotBehaviour.HasFlag(BotConfig.EBotBehaviour.MarkTradeMessagesAsRead) &&
+				message.StartsWith("[tradeoffer", StringComparison.Ordinal)) {
```

Comment:
It has a specific bbcode that, I believe, can't be sent manually (maybe we can send it with ASF, Archi know better, but I don't think it matters). So any trade message always starts with this, and regular message can never start with this (because `[` will be escaped with `\`), and so I guess it's a good test.

Reply:
This is fine, I don't think of any better solution to deal with this, and volvo bbcode is so specific that you can't just load it to some kind of parser.